src/passes.rs

```diff
@@ -33,7 +33,18 @@ impl<'a> Parser<'a> {
     pub fn new(text: &'a str) -> Parser<'a> {
         Parser::new_ext(text, Options::empty())
     }
-    pub fn new_ext(text: &'a str, mut opts: Options) -> Parser<'a> {
+    pub fn new_ext(text: &'a str, opts: Options) -> Parser<'a> {
+        Self::new_with_broken_link_callback(text, opts, None)
+    }
+
+    /// In case the parser encounters any potential links that have a broken
+    /// reference (e.g `[foo]` when there is no `[foo]: ` entry at the bottom)
+    /// the provided callback will be called with the reference name,
+    /// and the returned pair will be used as the link name and title if not
+    /// None.
+    pub fn new_with_broken_link_callback(text: &'a str, mut opts: Options,
```

Comment:
I'm curious whether the callback might be a reference rather than a `Box`, as it's only used for the lifetime of the parser (similar to `text`).

Reply:
It could, I didn't want to introduce another lifetime param. But reusing the same one may work.